modules/win_disable_services/manifests/disable_windows_update.pp

```diff
@@ -3,89 +3,97 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 class win_disable_services::disable_windows_update {
+  $win_update_key    = "HKLM\\SOFTWARE\\Policies\\Microsoft\\Windows\\WindowsUpdate"
+  $win_update_au_key = "${win_update_key}\\AU"
+  $win_au_key        = "HKLM\\SOFTWARE\\Microsoft\\Windows\\Windows\\AU"
+  service { 'wuauserv':
+    ensure => stopped,
+    name   => 'wuauserv',
+    enable => false,
+  }
+  case $facts['custom_win_release_id'] {
```

Comment:
Let's move away from using `$facts['custom_win_release_id']` . We can do this by updating the Powershell script at modules/win_shared/facts.d/facts_win_custom_os.ps1 to something like:

```
# Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

# Custom facts based off OS details that are not included in the default facts

# Windows release ID.
# From time to time we need to have the different releases of the same OS version
$release_key = (Get-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion')
$release_id = $release_key.ReleaseId
$win_os_build = [System.Environment]::OSVersion.Version.build

# OS caption
# Used to determine which KMS license for cloud workers
$caption = ((Get-WmiObject Win32_OperatingSystem).caption)
$caption = $caption.ToLower()
$os_caption = $caption -replace ' ', '_'
# Windows activation status
$status = (Get-CimInstance -ClassName SoftwareLicensingProduct -Filter "Name like 'Windows%'" | where PartialProductKey).licensestatus

If ($status -eq '1') {
$kms_status = "activated"
} else {
$kms_status = "needs_activation"
}

# Administrator SID
$administrator_info = Get-WmiObject win32_useraccount -Filter "name = 'Administrator'"
$win_admin_sid = $administrator_info.sid

# Network profile
# https://bugzilla.mozilla.org/show_bug.cgi?id=1563287
$NetCategory =  Get-NetConnectionProfile | select NetworkCategory

if ($NetCategory -like '*private*') {
	$NetworkCategory = "private"
} else {
	$NetworkCategory = "other"
}

# Firewall status

$firewall_status = (netsh advfirewall show domain state)

if ($firewall_status -like "*off*") {
	$firewall_status = "off"
} else {
	$firewall_status = "running"
}

# Base image ID
#$role = (Get-ItemProperty "HKLM:\SOFTWARE\Mozilla\ronin_puppet").role
$role = (Get-ItemProperty "HKLM:\SOFTWARE\Mozilla\ronin_puppet").role

# Get worker pool ID
$worker_pool_id = (Get-ItemProperty "HKLM:\SOFTWARE\Mozilla\ronin_puppet").worker_pool_id

if ($worker_pool_id -like "*gpu*") {
    $gpu = 'yes'
} else {
    $gpu = 'no'
}

if ($os_caption -like "*windows_10*") {
	$os_verision = (-join( "win_10_", $release_id))
} elseif ($os_caption -like "*windows_11*") {
	$os_verision = (-join( "win_11_", $release_id))
} elseif ($os_caption -like "*2012*") {
	$os_verision = win_11
} else {
	$os_verision = $null
}

write-host "custom_win_release_id=$release_id"
write-host "custom_win_os_caption=$os_caption"
write-host "custom_win_os_verision = $os_verision"
write-host "custom_win_kms_activated=$kms_status"
write-host "custom_win_admin_sid=$win_admin_sid"
Write-host "custom_win_net_category=$NetworkCategory"
Write-host "custom_win_firewall_status=$firewall_status"
Write-host "custom_win_role=$role"
write-host "custom_win_worker_pool_id=$worker_pool_id"
write-host "custom_win_gpu=$gpu"
```

This will allow us to the `$facts[custom_win_os_verision]` which in this case should have value of `win_11_2009`. This way it easier for people to identify what OS the configuration is for, and have the ability to differentiate between Win 10 and 11 that may have the same release ID.

Reply:
Sure. [Updated code.](https://github.com/mozilla-platform-ops/ronin_puppet/pull/407/commits/6e6c954e73c0908f2cf0636c324b64e8267e2eca)